### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by Dermiste**
One conditional include as to enable EUSART support in zephyr

### CODE REVIEW

**Review by jerome-pouiller — APPROVED** [review]
LGFM

**Comment by jerome-pouiller** [comment]
It seems I spoke too quickly. I don't find the corresponding symbol in the Zephyr repository. Do you have a PR to submit to Zephyr core project before this PR?

**Review by rzr — APPROVED** [review]
May commit message improved to tell more about context something like:

 gecko: Add usart to build recipe

**Comment by Dermiste** [comment]
There is no PR on Zephyr side, but I thought that would be an issue since this new line is kind of a loose addition, i.e won't cause any issue and make the code base ready for a symbol that will exist at some point?

**Comment by jerome-pouiller** [comment]
If `em_eusart.c` is orphan, we should probably drop it. However, I assume we keep it because `emlib/` is a raw copy from the GSDK and we don't want to diverge.

I would tend to not apply this patch until we have support for EUSART in Zephyr-core. However, it is not a strong opinion.

**Comment by fkokosinski** [comment]
> There is no PR on Zephyr side, but I thought that would be an issue since this new line is kind of a loose addition, i.e won't cause any issue and make the code base ready for a symbol that will exist at some point?

Hi. We would need you to create a PR to Zephyr that bumps the version of the SiLabs HAL to verify your changes in a automated way. This is actually a requirement that the Zephyr Project sets for us itself: https://docs.zephyrproject.org/latest/develop/modules.html#submitting-changes-to-modules.

**Comment by Terezvent** [comment]
Hi everyone, I'm working on adding support for SPI over EUSART in zephyr, I was about to send a PR with the same changes, I'd like to know if this PR will be approved or should I send my own?

**Comment by jerome-pouiller** [comment]
This PR cannot be approved as-is because the introduced symbol is orphan. I suggest to send your own PR (then I will mark this PR superseded).

**Comment by Terezvent** [comment]
Ok, thanks @jerome-pouiller!

**Comment by Dermiste** [comment]
I think there is an open PR on the remote library, I'm not working anymore
on that thing so no idea if it was merged at some point or not, maybe just
changing the version of the lib will make the orphan symbol not orphan
anymore

File: gecko/CMakeLists.txt [hunk 1]
```diff
@@ -175,6 +175,7 @@ zephyr_sources_ifdef(CONFIG_COUNTER_GECKO_STIMER   service/sleeptimer/src/sl_sle
 zephyr_sources_ifdef(CONFIG_SOC_GECKO_SE           emlib/src/em_se.c)
 zephyr_sources_ifdef(CONFIG_SOC_GECKO_TIMER        emlib/src/em_timer.c)
 zephyr_sources_ifdef(CONFIG_SOC_GECKO_USART        emlib/src/em_usart.c)
+zephyr_sources_ifdef(CONFIG_SOC_GECKO_EUSART       emlib/src/em_eusart.c)
```
Comment (rzr) [inline gecko/CMakeLists.txt:178]: I would sort entries alphabetically but it is not a big deal

**Review by rzr — APPROVED** [review]

**Comment by asmellby** [comment]
Superseded by #67 and subsequently #68 .

**Pull request closed without merging**